src/pyqasm/pulse/pulse_validator.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2025 qBraid
```

Comment:
Nitpick : rename to `pulse/validator.py` as we are already inside the `pulse` sub-package